src/main/java/emissary/output/DropOffUtil.java

```diff
@@ -997,6 +1009,31 @@ void extractUniqueFileExtensions(IBaseDataObject p) {
         }
     }
 
+    /**
+     * Checks the Original-Filename and FILE_ABSOLUTEPATH for the filename of the object. Returns a list with of the
+     * non-empty strings found in these fields. If nothing is found in either field, return an empty list.
+     *
+     * @param d The IBDO
+     * @return The list of filenames found in the field Original-Filename or FILE_ABSOLUTEPATH
+     */
+    public static List<String> getBestFilenames(IBaseDataObject d) {
+        String[] fieldsToTry = {ORIGINAL_FILENAME, FILE_ABSOLUTEPATH};
+
+        List<String> filenames = new ArrayList<>();
+
+        for (String ibdoField : fieldsToTry) {
+            if (d.hasParameter(ibdoField)) {
+                for (Object filename : d.getParameter(ibdoField)) {
+                    String stringFileName = (String) filename;
+                    if (StringUtils.isNotEmpty(stringFileName)) {
```

Comment:
`isNotBlank` will also protect against null values.  Should be able to explicitly declare in the method's contract (documented in Javadoc) that the returned List will only return non-blank values, so callers will know they can skip null-checks on the returned values.
```suggestion
                    if (StringUtils.isNotBlank(stringFileName)) {
```